pkg/controllers/nodeclass/status/validation_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status_test
+
+import (
+	status "github.com/awslabs/operatorpkg/status"
+	"github.com/samber/lo"
+
+	v1 "github.com/aws/karpenter-provider-aws/pkg/apis/v1"
+	"github.com/aws/karpenter-provider-aws/pkg/test"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "sigs.k8s.io/karpenter/pkg/test/expectations"
+)
+
+var _ = Describe("NodeClass Validation Status Controller", func() {
+	BeforeEach(func() {
+		nodeClass = test.EC2NodeClass(v1.EC2NodeClass{
+			Spec: v1.EC2NodeClassSpec{
+				SubnetSelectorTerms: []v1.SubnetSelectorTerm{
+					{
+						Tags: map[string]string{"*": "*"},
+					},
+				},
+				SecurityGroupSelectorTerms: []v1.SecurityGroupSelectorTerm{
+					{
+						Tags: map[string]string{"*": "*"},
+					},
+				},
+				AMIFamily: lo.ToPtr(v1.AMIFamilyCustom),
+				AMISelectorTerms: []v1.AMISelectorTerm{
+					{
+						Tags: map[string]string{"*": "*"},
+					},
+				},
+				Tags: map[string]string{
```

Comment:
Should we create tests that try this for all restricted tag key patterns? Maybe something that does a DescribeTable and iterates through all of the invalid options?

Reply:
Done.